client/service/src/lib.rs

```diff
@@ -346,45 +333,52 @@ where
 		.rpc_ws
 		.unwrap_or_else(|| "127.0.0.1:9944".parse().expect("valid sockaddr; qed"));
 	let ws_addr2 = random_port(ws_addr);
+
 	let http_addr = config
 		.rpc_http
 		.unwrap_or_else(|| "127.0.0.1:9933".parse().expect("valid sockaddr; qed"));
 	let http_addr2 = random_port(http_addr);
 
 	let metrics = sc_rpc_server::RpcMetrics::new(config.prometheus_registry())?;
 
-	let http_fut = sc_rpc_server::start_http(
-		[http_addr, http_addr2],
-		config.rpc_cors.as_ref(),
-		max_request_size,
-		http_max_response_size,
-		metrics.clone(),
-		gen_rpc_module(deny_unsafe(ws_addr, &config.rpc_methods))?,
-		config.tokio_handle.clone(),
-	);
-
-	let ws_config = WsConfig {
+	let server_config = sc_rpc_server::Config {
 		max_connections: config.rpc_ws_max_connections,
 		max_payload_in_mb: max_request_size,
 		max_payload_out_mb: ws_max_response_size,
 		max_subs_per_conn: config.rpc_max_subs_per_conn,
 	};
 
-	let ws_fut = sc_rpc_server::start_ws(
+	let ws_fut = sc_rpc_server::start_server(
 		[ws_addr, ws_addr2],
 		config.rpc_cors.as_ref(),
-		ws_config,
+		server_config.clone(),
+		metrics.clone(),
+		gen_rpc_module(deny_unsafe(ws_addr, &config.rpc_methods))?,
+		config.tokio_handle.clone(),
+		rpc_id_provider,
+		"WS",
+	);
+
+	let http_fut = sc_rpc_server::start_server(
```

Comment:
NOTE: for backwards compatibility just to start another server on port 9333 i.e, `--rpc-port` I would want to kill remove it completely but that would break lots of things :P

Reply:
I could also use the old http server i.e, jsonrpsee v0.15 but I would prefer not to.